### PR TITLE
Fix stuff and make it faster

### PR DESCRIPTION
**Pull request by stefanpenner**
* upgrade broccoli-caching-writer
* fix issue where inputTrees are being overwritten by merge

- [x] fix perf regression by releasing broccoli-caching-writer v2.1.0

### CODE REVIEW

**Comment by jasonmit** [comment]
Thanks!

File: index.js [hunk 1]
```diff
@@ -1,26 +1,26 @@
...
-var writeFile = RSVP.denodeify(fs.writeFile)
+var fs = require('fs');
+var path = require('path');
+var mkdirp = require('mkdirp');
+var includePathSearcher = require('include-path-searcher');
+var CachingWriter = require('broccoli-caching-writer');
+var less = require('less');
+var merge = require('lodash.merge');
+var RSVP = require('rsvp');
+var writeFile = RSVP.denodeify(fs.writeFile);
```
Comment (jasonmit) [inline index.js]: No longer used it looks, I'll remove before versioning

File: index.js [hunk 2]
```diff
@@ -1,26 +1,26 @@
...
-var RSVP = require('rsvp')
-var writeFile = RSVP.denodeify(fs.writeFile)
+var fs = require('fs');
+var path = require('path');
+var mkdirp = require('mkdirp');
+var includePathSearcher = require('include-path-searcher');
+var CachingWriter = require('broccoli-caching-writer');
+var less = require('less');
+var merge = require('lodash.merge');
+var RSVP = require('rsvp');
```
Comment (jasonmit) [inline index.js]: Same as below

File: index.js (hunk 1 above)
Comment (stefanpenner) [inline index.js]: :+1:

File: index.js (hunk 2 above)
Comment (stefanpenner) [inline index.js]: :+1:

**Comment by stefanpenner** [comment]
For reference it appears (in a large app you may be familiar with) the cache key construction is about 2x as fast. It also fixes issues with invalidation from unexpected things, which means it should reduce the chances of a rebuild for silly reasons, which is likely much more then a 2x win.

**Comment by jasonmit** [comment]
Awesome work as always, thank you and just published as 0.6.0

**Pull request merged**